CRM/Civigiftaid/Utils/Contribution.php

```diff
@@ -395,13 +400,16 @@ public static function validateContributionToBatch(&$contributionIDs) {
     require_once "CRM/Contribute/BAO/Contribution.php";
 
     foreach ($contributionIDs as $contributionID) {
-      $batchContribution = new CRM_Batch_DAO_EntityBatch();
-      $batchContribution->entity_table = 'civicrm_contribution';
-      $batchContribution->entity_id = $contributionID;
+      $contributionInBatch = civicrm_api3('EntityBatch', 'get', [
+          'sequential' => 1,
+          'entity_table' => "civicrm_contribution",
+          'batch_id.type_id' => "giftaid_batch",
+          'entity_id' => $contributionID,
+      ]);
```

Comment:
We can use the same function here too. The function that fetches contributions in batch with a single API call and we can check the array returned to determine if contribution is in batch or not.